src/settings/LanguagePicker.js

```diff
@@ -15,26 +19,57 @@ const styles = StyleSheet.create({
 type Props = {|
   value: string,
   onValueChange: (locale: string) => void,
+  filter: string,
+  _: GetText,
 |};
 
-export default class LanguagePicker extends PureComponent<Props> {
+class LanguagePicker extends PureComponent<Props> {
+  getTranslatedLanguages = () =>
+    languages.map(language => {
+      const translatedName = this.props._(language.name);
+      return {
+        ...language,
+        name: translatedName,
+      };
+    });
+
+  getFilteredLanguageList = (filter: string) => {
+    const list = this.getTranslatedLanguages();
+
+    if (!filter) {
+      return list;
+    }
+
+    return list.filter(item => {
+      const itemData = `${item.name.toUpperCase()} ${item.nativeName.toUpperCase()}`;
+      const filterData = filter.toUpperCase();
+
+      return itemData.includes(filterData);
+    });
+  };
+
   render() {
-    const { value, onValueChange } = this.props;
+    const { value, onValueChange, filter } = this.props;
+    const data = this.getFilteredLanguageList(filter);
 
     return (
       <FlatList
         ItemSeparatorComponent={() => <View style={styles.separator} />}
         initialNumToRender={languages.length}
-        data={languages}
+        data={data}
         keyExtractor={item => item.locale}
         renderItem={({ item }) => (
           <LanguagePickerItem
             selected={item.locale === value}
             onValueChange={onValueChange}
-            {...item}
+            locale={item.locale}
+            name={item.name}
+            nativeName={item.nativeName}
           />
         )}
       />
     );
   }
 }
+
+export default connect()(withGetText(LanguagePicker));
```

Comment:
You don't need `connect` to use `withGetText`.
We use it to extract data out of the Redux store and/or to get the `dispatch` function.

Reply:
Hmm interesting. I wrote `withGetText`, recently; I agree with this:

> Logically, `withGetText` will inject prop `_` to `LanguagePicker`, so there is no need to pass `_` here.

and I think I wrote its Flow type wrong. :grin: 

I'll look at that and try to fix it, but in the meantime there's an alternate way to use our translation API that might be a little nicer here anyway. We don't have any examples of this yet because the API is quite new, but here's what it looks like at the tip of this PR branch:
```
@@ -4,7 +4,7 @@ import { FlatList, StyleSheet, View } from 'react-native';
 import { connect } from 'react-redux';
 
 import type { GetText } from '../types';
-import { withGetText } from '../boot/TranslationProvider';
+import { withGetText, TranslationContext } from '../boot/TranslationProvider';
 
 import languages from './languages';
 import LanguagePickerItem from './LanguagePickerItem';
@@ -20,13 +20,16 @@ type Props = {|
   value: string,
   onValueChange: (locale: string) => void,
   filter: string,
-  _: GetText,
 |};
 
-class LanguagePicker extends PureComponent<Props> {
+export default class LanguagePicker extends PureComponent<Props> {
+  static contextType = TranslationContext;
+  context: GetText;
+
   getTranslatedLanguages = () =>
     languages.map(language => {
-      const translatedName = this.props._(language.name);
+      const _ = this.context;
+      const translatedName = _(language.name);
       return {
         ...language,
         name: translatedName,
@@ -71,5 +74,3 @@ class LanguagePicker extends PureComponent<Props> {
     );
   }
 }
-
-export default connect()(withGetText(LanguagePicker));
```

Try that way?

Reply:
One quick important workflow point, on this topic: when you fix something, use `git rebase -i` to squash it into the commit that it's fixing. That's part of how we make [clear and coherent commits](https://zulip.readthedocs.io/en/latest/contributing/version-control.html).

Reply:
Thanks for the suggestion, that works perfectly :) 

As a side note, I also tried following:
```
/**
 * Usually called `_`, and invoked like `_('Message')` -> `'Nachricht'`.
 *
 * Use `context: TranslationContext` in a React component; then in methods,
 * say `const _ = this.context`.
 *
 * Alternatively, for when `context` is already in use, use `withGetText`
 * and then say `const { _ } = this.props`.
 *
 * @prop intl - The full react-intl API, for more complex situations.
 */
export type GetText = {
  (string): string,
  intl: IntlShape,
};
```
And it doesn't work. I think the comment may be misleading here.

Reply:
Also, I've used `rebase` to squash commits. Thanks for suggesting!

Reply:
Yeah, indeed that comment is mistaken; I'll fix it to match the instructions I gave above (and that you've just validated actually work :slightly_smiling_face: ). Thanks!